techmatters/terraso-backend
# Feat/permissions improvements

## Description

Rewrite LandPKS project management & soil data permissions system to comply with the [product-approved requirements document](https://airtable.com/appDU4llBX3AicyTX/tblo1pU8zkQzRmFsF/viwt0gf1rs4e7130i?blocks=hide). Add new unit and integration testing where appropriate.

The new system is built around permission groups derived directly from the product team's requirements. This is paired with a table that maps action names to these groups, accounting for contextual conditions such as affiliated / unaffiliated sites. The new system uses a unified context model for permission decisions, which formalizes a pattern which was previously emerging in the prior implementation.

### Checklist

### Related Issues
Fixes [#595](https://github.com/techmatters/terraso-product/issues/595)

### Verification steps

Run backend test suite. Explore application and verify that permissions system still behaves as expected.
